File: ruby/src/Behavioral/State/DvdStateNameStars.rb
# DvdStateNameStars- two of two Concrete States

require "Behavioral/State/DvdStateName"
require "Behavioral/State/DvdStateNameExclaim"

class DvdStateNameStars < DvdStateName

	def initialize(starCount = 0)
		@starCount = starCount
	end
	
	def showName (ctx, name)
		@starCount += 1
		retval = name.sub(' ','*')  #// show stars twice, switch to exclamation point
		if @starCount > 1
 			ctx.setDvdStateName( DvdStateNameExclaim.new )
		end
		retval
	end
end

File: java/src/Behavioral/State/DvdStateNameStars.java
package Behavioral.State;


//DvdStateNameStars.java - two of two Concrete States

public class DvdStateNameStars implements DvdStateName {  
    int starCount;
    
    public DvdStateNameStars() {
        starCount = 0;
    }
    
    public String showName(DvdStateContext dvdStateContext, 
                         String nameIn) {
         String retval = nameIn.replace(' ','*');
         // show stars twice, switch to exclamation point
         if (++starCount > 1) {
             dvdStateContext.setDvdStateName( new DvdStateNameExclaim());
         }       
			return retval;
    }  
}
